File: basic_scala/src/main/scala/pers/xiaoming/spark/core/StudentScore.scala
package pers.xiaoming.spark.core

class StudentScore(val name:String,
                   private val total:Int,
                   private val math:Int,
                   private val english:Int) extends Ordered[StudentScore] with Serializable {

  override def compare(that: StudentScore): Int = {
    if (this.total - that.total != 0) {
      this.total - that.total
    } else {
      this.math - that.math
    }
  }

  override def toString = s"StudentScore($name, $total, $math, $english)"
}

File: basic_java/src/main/java/pers/xiaoming/spark/core/StudentScore.java
package pers.xiaoming.spark.core;

import java.io.Serializable;

public class StudentScore implements Serializable {
    String name;
    int total;
    int math;
    int english;

    public StudentScore(String name, int total, int math, int english) {
        this.name = name;
        this.total = total;
        this.math = math;
        this.english = english;
    }

    public String getName() {
        return name;
    }

    @Override
    public String toString() {
        return "StudentScore{" +
                "name='" + name + '\'' +
                ", total=" + total +
                ", math=" + math +
                ", english=" + english +
                '}';
    }
}
